source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -274,6 +271,19 @@ void DnsCacheImpl::onReResolve(const std::string& host) {
   }
 }
 
+void DnsCacheImpl::forceRefreshHosts() {
+  absl::ReaderMutexLock reader_lock{&primary_hosts_lock_};
+  for (auto& primary_host : primary_hosts_) {
+    // Avoid holding the lock for longer than necessary by just triggering the refresh timer for
+    // each host IFF the host is not already refreshing.
```

Comment:
I'm all for minimizing lock time, but if we've started a resolution for www.lyft.com when we were on wifi and just switched to cellular and wifi is donw, that result would then get blackholed and we really do want to kick off a new attempt, no?
If you think it's not worth it because that resolve will time out, and force reresolve, mind adding the expected failure path to the comment here?

Reply:
I was mostly just trying to keep it simple for right now. I agree that it would be better to cancel the in flight attempt and start a new one. I can either add a TODO about that or just do it now? Preference?

Reply:
either a TODO or just an updated comment noting the best-effort nature is fine :-)